code/obj/machinery/computer.dm

```diff
@@ -12,19 +12,77 @@
 	/// does it have a glow in the dark screen? see computer_screens.dmi
 	var/glow_in_dark_screen = TRUE
 	var/image/screen_image
+
+	var/can_reconnect = 0 //Set to 1 to make multitools call connection_scan. For consoles with associated equipment (cloner, genetek etc)
+	var/obj/item/circuitboard/circuit_type = null
+	Topic(href, href_list)
+		if (..(href, href_list))
+			return 1
+		playsound(src.loc, 'sound/machines/keypress.ogg', 30, 1, -15)
+
+	attack_hand(var/mob/user)
+		..()
+		if (!user.literate)
+			boutput(user, "<span class='alert'>You don't know how to read or write, operating a computer isn't going to work!</span>")
+			return 1
+		interact_particle(user,src)
+
+	attack_ai(mob/user as mob)
+		src.attack_hand(user)
+
+	attackby(obj/item/W as obj, mob/user as mob)
+		if (can_reconnect)
+			if (istype(W, /obj/item/device/multitool) && !(status & (BROKEN|NOPOWER)))
+				boutput(user, "<span class='notice'>You pulse [src.name] to re-scan for equipment.</span>")
+				connection_scan()
+				return
+		if (isscrewingtool(W) && src.circuit_type)
+			playsound(src.loc, "sound/items/Screwdriver.ogg", 50, 1)
+			if (do_after(user, 2 SECONDS))
```

Comment:
Outside scope of this PR, but probably a good candidate for an action bar.

Reply:
I can try that as a follow up PR.